### PR TITLE
drivers: entropy: Enable entropy on RT11XX

### PR DESCRIPTION
**Pull request by decsny**
Enable entropy driver on MIMXRT11XX using on-chip CAAM RNG 
Update docs to reflect
Adds CAAM to mcux configurations (DT and Kconfig)

### CODE REVIEW

File: dts/arm/nxp/nxp_rt11xx.dtsi [hunk 1]
```diff
@@ -699,6 +703,14 @@
 			};
 		};
 
+		caam: caam@40440000 {
+			compatible = "nxp,imx-caam";
+			reg = <0x40440000 0x81000>;
+			interrupts = <69 70 71 72 73 74>;
+			status = "okay";
+			label = "CAAM";
```
Comment (galak) [inline dts/arm/nxp/nxp_rt11xx.dtsi:711]: remove label property.

File: soc/arm/nxp_imx/rt/Kconfig.defconfig.series [hunk 2]
```diff
@@ -24,6 +24,10 @@ config ENTROPY_MCUX_TRNG
 	default y if HAS_MCUX_TRNG
 	depends on ENTROPY_GENERATOR
 
+config ENTROPY_MCUX_CAAM
```
Comment (galak) [inline soc/arm/nxp_imx/rt/Kconfig.defconfig.series:27]: remove this, the driver will get default y if its enabled in devicetree so this isn't needed.

**Review by galak — CHANGES_REQUESTED** [review]
Some minor Kconfig & dtsi changes needed.

File: modules/Kconfig.mcux [hunk 3]
```diff
@@ -229,6 +229,12 @@ config HAS_MCUX_TRNG
 	  Set if the true random number generator (TRNG) module is present in
 	  the SoC.
 
+config HAS_MCUX_CAAM
```
Comment (danieldegrasse) [inline modules/Kconfig.mcux:232]: The `HAS_MCUX_CAMM` symbol should not be needed with the `DT_HAS` method of enabling device drivers

**Review by galak — CHANGES_REQUESTED** [review]
interrupt property in dtsi doesn't look correct.

File: dts/arm/nxp/nxp_rt11xx.dtsi [hunk 4]
```diff
@@ -699,6 +703,13 @@
 			};
 		};
 
+		caam: caam@40440000 {
+			compatible = "nxp,imx-caam";
+			reg = <0x40440000 0x81000>;
+			interrupts = <69 70 71 72 73 74>;
```
[the remark this answers is not in the document]
Comment (decsny) [reply dts/arm/nxp/nxp_rt11xx.dtsi]: thank you for that catch

**Review by danieldegrasse — DISMISSED** [review]

File: west.yml [hunk 5]
```diff
@@ -93,7 +93,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 054460d6725c2db5cddcdddd0c5bbe6caeae5d36
+      revision: pull/178/head
```
Comment (mmahadevan108) [inline west.yml:96]: Please update this to point to the commit from the `NXP HAL` repo

File: dts/bindings/rng/nxp,imx-caam.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2018, Linaro Limited
```
Comment (mmahadevan108) [inline dts/bindings/rng/nxp,imx-caam.yaml:1]: Please update the copyright

File: drivers/entropy/entropy_mcux_caam.c [hunk 7]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2017 Linaro Limited.
```
Comment (mmahadevan108) [inline drivers/entropy/entropy_mcux_caam.c:2]: Please update the copyright

File: drivers/entropy/entropy_mcux_caam.c [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+	.get_entropy = entropy_mcux_caam_get_entropy
+};
+
+static const struct mcux_entropy_config entropy_mcux_config = {
+	.base = (CAAM_Type *)DT_INST_REG_ADDR(0)
+};
+
+static int entropy_mcux_caam_init(const struct device *);
+
+DEVICE_DT_INST_DEFINE(0,
```
Comment (galak) [inline drivers/entropy/entropy_mcux_caam.c:52]: Just move this down, so you can remove the deceleration of `entropy_mcux_caam_init`

File: drivers/entropy/entropy_mcux_caam.c [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+		    PRE_KERNEL_1, CONFIG_ENTROPY_INIT_PRIORITY,
+		    &entropy_mcux_caam_api_funcs);
+
+static int entropy_mcux_caam_init(const struct device *dev)
+{
+	const struct mcux_entropy_config *config = dev->config;
+	caam_config_t conf;
+	status_t status;
+
+	ARG_UNUSED(dev);
```
Comment (galak) [inline drivers/entropy/entropy_mcux_caam.c:64]: remove ARG_UNUSED(dev);

**Review by galak — DISMISSED** [review]
a few code changes in the driver.

File: west.yml [hunk 10]
```diff
@@ -93,7 +93,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 9512229c35e230003c64fda41be578f2718b70f4
+      revision: 90db601fa84d203fd6d731cba0e391016b66e319
```
Comment (mmahadevan108) [inline west.yml:96]: Check this commit hash value. This is causing CI failures

Comment (decsny) [reply west.yml:96]: I also noticed the west setup failing on updating nxp hal earlier and I double checked this commit hash twice now, I don't know what the problem is yet though

**Review by mmahadevan108 — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**